scene/2d/camera_2d.cpp

```diff
@@ -409,6 +409,35 @@ void Camera2D::force_update_scroll() {
 	_update_scroll();
 }
 
+void Camera2D::reset_smoothing() {
+
+	smoothed_camera_pos = camera_pos;
+	_update_scroll();
+}
+
+void Camera2D::align() {
+
+	Size2 screen_size = get_viewport_rect().size;
+	screen_size=get_viewport_rect().size;
+	Point2 current_camera_pos = get_global_transform().get_origin();
+	if (anchor_mode==ANCHOR_MODE_DRAG_CENTER) {
+		if (h_ofs<0) {
```

Comment:
It's exactly the same code like in `get_camera_transform()`. I first thought of capsuling the code in a function so we don't have code duplicates, but I did't find a proper name for it. And all along it uses a stack of variables, that are going to be changed in the function, so I think it's okay here to have this code or rather its functionality duplicated.